src/coreclr/vm/syncblk.cpp

```diff
@@ -2549,14 +2554,9 @@ BOOL AwareLock::EnterEpilogHelper(Thread* pCurThread, INT32 timeOut)
         QueryPerformanceCounter(&startTicks);
 
         // Fire a contention start event for a managed contention
-        FireEtwContentionStart_V1(ETW::ContentionLog::ContentionStructs::ManagedContention, GetClrInstanceId());
+        FireEtwContentionStart_V2(ETW::ContentionLog::ContentionStructs::ManagedContention, GetClrInstanceId(), OBJECTREFToObject(obj), m_HoldingThread);
```

Comment:
Rather than writing out `m_HoldingThread` which is the address of the `Thread` object in memory, can we change this to `m_HoldingThread->GetOSThreadId64()`?  This will allow a profiler to display what the thread that currently holds the lock is doing.  With the pointer value, you would need to provide another event that correlates between `Thread*` and the actual OS thread ID.

Reply:
Another potential issue here is that the value of `obj` can change during GC relocation.  Is the goal to also watch the GC survival and movement events, or do you want to try and have a more durable ID?  I'm wondering if `AwareLock` should also have a set of creation/destruction events so that you can see the full lifetime of the lock.

Reply:
@brianrob good point. I'll make the change to replace the pointer by the OS Thread id.

> Another potential issue here is that the value of obj can change during GC relocation.

Does it mean that the address is changed also ?
The goal, at least in this PR, is not to watch the lock object lifetime, but more having an ID to do some correlation.

Reply:
> Does it mean that the address is changed also ?

Yes the object address also changes when it's moved. Could pass in `this` instead to include the `AwareLock` pointer as just a unique identifier for the lock.

Reply:
Using `this` is a good idea.  Worth calling out though that it is possible to destroy and then create a new lock at the same address.  To make sure that you know about this, you may wish to write an event in the `AwareLock` ctor so that you know when the aware lock is created.  This type of pattern is other places such as module and type loads.  We use the address of the module or type in memory, but handle questions of lifetime using load/unload events.

Reply:
Thanks for the info.
So if I understand correctly, in the `AwareLock` ctor I emit an event (we know the lock is created) and I will have to emit an event in the dtor to know when the lock is deleted, right ?

I'll do the change taking into account your comment.

Reply:
To avoid emitting too many events, I think you can safely emit just the creation event.  If you see two creation events at the same address, you know that it's a new lock.